qpid-jms-client/src/test/java/org/apache/qpid/jms/test/testpeer/TestAmqpPeer.java

```diff
@@ -495,6 +512,169 @@ public void run()
         }
     }
 
+    public void expectGSSAPIFail(Symbol mech) throws Exception {
+        SaslMechanismsFrame saslMechanismsFrame = new SaslMechanismsFrame().setSaslServerMechanisms(mech);
+
+        addHandler(new HeaderHandlerImpl(AmqpHeader.SASL_HEADER, AmqpHeader.SASL_HEADER,
+                new FrameSender(
+                        this, FrameType.SASL, 0,
+                        saslMechanismsFrame, null)));
+
+        addHandler(new SaslInitMatcher().withMechanism(equalTo(mech)));
+
+    }
+
+    public void expectGSSAPI(Symbol mech, String serviceName) throws Exception {
+
+        SaslMechanismsFrame saslMechanismsFrame = new SaslMechanismsFrame().setSaslServerMechanisms(mech);
+
+        addHandler(new HeaderHandlerImpl(AmqpHeader.SASL_HEADER, AmqpHeader.SASL_HEADER,
+                new FrameSender(
+                        this, FrameType.SASL, 0,
+                        saslMechanismsFrame, null)));
+
+        // setup server gss context
+        LoginContext loginContext = new LoginContext("", null, null,
+                kerb5InlineConfig(serviceName, false));
```

Comment:
I have added some unverified support for ibm jdk - I guess it is an argument for only having external jaas config via login.config. But I think in this case ease of use trumps the overhead of maintaining the inline config